cms/djangoapps/contentstore/views/preview.py

```diff
@@ -296,10 +297,10 @@ def _studio_wrap_xblock(xblock, view, frag, context, display_name_only=False):
             'is_root': is_root,
             'is_reorderable': is_reorderable,
             'can_edit': context.get('can_edit', True),
-            'can_edit_visibility': context.get('can_edit_visibility', True),
+            'can_edit_visibility': context.get('can_edit_visibility') or is_not_library,
```

Comment:
Maybe change this to `context.get('can_edit_visibility', not is_library)` and add an explanatory comment?